internal/serve/httphandler/receiver_send_otp_handler_test.go

```diff
@@ -26,9 +27,11 @@ import (
 
 type mockMessengerClient struct {
 	mock.Mock
+	message message.Message
 }
 
 func (m *mockMessengerClient) SendMessage(message message.Message) error {
+	m.message = message
```

Comment:
💡  I know this approach works, but this is an anti-pattern because it doesn't take advantage of the mock library very well. Since we use testify's mocks, we should rely on the testing methods for the assertions.

Something like this could help you get a copy of that object for your assertions:
```go
		var receivedMessage message.Message
		mockMessenger.On("SendMessage", mock.AnythingOfType("message.Message")).
			Return(nil).
			Once().
			Run(func(args mock.Arguments) {
				receivedMessage = args.Get(0).(message.Message)
			})
```

Reply:
Wooooowww this is awesome!! Thank you so much.